### ISSUE TITLE
Finish implementing job queue

### ISSUE THREAD

**Issue by ufdada** [tonihele/OpenKeeper#224]
Toni already made a nice implementation of digging and idling, now the rest should be done.

DK2 has some nice references hidden how they were called internally.
Since some tasks may be a little to big, we can make them one by one. We just mark whats done so far.

Here is a short list:
- [ ] CallToArmsState
- [ ] CasinoState
- [ ] CombatPitState
- [ ] DisArmTrapState
- [ ] ExploreState
- [ ] FightWithDrawState
- [ ] GoodState //?
- [ ] GuardState
- [ ] HungerState
- [x] ImpState // i guess this is digging
- [ ] KeeperState
- [x] LairState
- [ ] ManufactureState
- [ ] ObjectState //?
- [ ] PatrolState
- [ ] PayState // Payday!
- [ ] PrayState
- [ ] PrisonState
- [ ] ReaperState
- [ ] ResearchState
- [ ] RunAway
- [ ] SulkState // probably entering when the dungeon doesn't fullfil their needs, then going back to portal
- [ ] ThrowingState
- [ ] TourtureState
- [ ] TrainState
- [x] WaitState // probably idle

DK2 also seems to have some decent debugging for these states, but i didn't found out yet how to activate it.

**Comment by tonihele** [comment]
Just to note that currently some these aren't really states per se... Many of these has been implemented as tasks. For me at least it makes a small difference.

**Comment by ufdada** [comment]
Well it's just what Bullfrog used as names in the original Engine (you can find them in the executable). I don't think we need to stick with the names if there is no reason for it.

**Comment by tonihele** [comment]
Yes and that is a good list. I just wanted to make clear, since we have states & tasks. States as in state machine and WorkState that performs tasks. The design might differ a lot from the original. And it may be that we should have states as opposed to tasks for these, or not.
